sprangu-back/src/main/java/com/sprangu/spranguback/domain/user/UserService.java

```diff
@@ -1,36 +1,66 @@
 package com.sprangu.spranguback.domain.user;
 
+import com.sprangu.spranguback.application.config.JwtUtils;
 import com.sprangu.spranguback.domain.user.model.dto.UserInfo;
 import com.sprangu.spranguback.domain.user.model.entity.RegisteredUser;
+import com.sprangu.spranguback.domain.user.model.security.UserDetailed;
 import com.sprangu.spranguback.domain.user.repository.UserRepository;
 import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.stereotype.Service;
 
+import javax.security.auth.login.CredentialException;
 import javax.transaction.Transactional;
+import java.util.Collections;
 
 @Service
 @AllArgsConstructor
 @Transactional
-public class UserService {
+public class UserService implements UserDetailsService {
 
     private final UserRepository userRepository;
+    private final JwtUtils jwtUtils;
 
+    public UserDetailed getLoggedUser(){
+        var auth = SecurityContextHolder.getContext().getAuthentication();
+        return (UserDetailed) auth.getPrincipal();
```

Comment:
O kodėl ne one-liner'is? Jeigu eit šituo keliu tai gal jau iki galo? 😃

Reply:
Tbh niekur :D palikau, kad jeigu kažkam reiktų tiesiog galėtų šitą pacallinit, nes gali būt net nežinomas toks funkcionalumas, kad galima current useri gauti